tests/packagedcode/test_npm.py

```diff
@@ -245,6 +245,19 @@ def test_parse_npm_shrinkwrap(self):
         packages = npm.parse(test_file)
         self.check_packages(packages, expected_loc, regen=False)
 
+    def test_parse_with_name(self):
+        test_file = self.get_test_loc('npm/with_name/package.json')
+        expected_loc = self.get_test_loc('npm/with_name/package.json.expected')
+        packages = npm.parse(test_file)
+        self.check_packages(packages, expected_loc, regen=False)
+
+    def test_parse_without_name(self):
+        test_file = self.get_test_loc('npm/without_name/package.json')
+        try:
+            npm.parse(test_file)
+        except AttributeError as e:
+            assert "'NoneType' object has no attribute 'to_dict'" in str(e)
```

Comment:
I guess there be a better exception raises, like a new one such as a `NpmWithoutNameError` ? 
But we can merge this as is for now. An update later is welcomed!